internal/services/orbital/contact_profile_resource.go

```diff
@@ -0,0 +1,279 @@
+package orbital
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/orbital/2022-03-01/contactprofile"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/network/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ContactProfileResource struct{}
+
+var _ sdk.ResourceWithUpdate = ContactProfileResource{}
+
+type ContactProfileResourceModel struct {
+	Name                           string                    `tfschema:"name"`
+	ResourceGroup                  string                    `tfschema:"resource_group_name"`
+	Location                       string                    `tfschema:"location"`
+	MinimumVariableContactDuration string                    `tfschema:"minimum_variable_contact_duration"`
+	MinimumElevationDegrees        float64                   `tfschema:"minimum_elevation_degrees"`
+	AutoTrackingConfiguration      string                    `tfschema:"auto_tracking"`
+	EventHubUri                    string                    `tfschema:"event_hub_uri"`
+	Links                          []ContactProfileLinkModel `tfschema:"links"`
+	Tags                           map[string]string         `tfschema:"tags"`
+	NetworkConfigurationSubnetId   string                    `tfschema:"network_configuration_subnet_id"`
+}
+
+func (r ContactProfileResource) Arguments() map[string]*schema.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": azure.SchemaResourceGroupName(),
+
+		"location": azure.SchemaLocation(),
+
+		"links": ContactProfileLinkSchema(),
```

Comment:
is there a reason you are pulling these out?

Reply:
Yes. This is schema has a nested structure, I'm pulling it out for better code readability.

Reply:
unless it is reused we should not be doing this